### ISSUE TITLE
JS: False-positive for undefined variable inside a returned object

### ISSUE THREAD

**Issue by aireol** [SonarSource/SonarJS#1332]
**I want to report a bug.**

**SonarJS version:** 5.1 (as of SonarLint 1.6.0)<br>
**SonarLint version:** 1.10.0<br>

**Reproducer** 
```
function foo() {
    return {
        bar: buzz
    }
}
```
**Expected behavior**
```
function foo() {
    return {
        bar: buzz // -->is buzz should be reported/highlighted as an undefined variable?
    }
}
```
